service/src/test/java/org/sunbird/learner/actors/skill/UserSkillManagementActorTest.java

```diff
@@ -0,0 +1,373 @@
+  private static final String SKILL_NAME = "Java";
+  private static final String ENDORSE_SKILL_NAME = "C";
+  private static List<String> skillsList = new ArrayList<>();
+  private Util.DbInfo userSkillDbInfo = Util.dbInfoMap.get(JsonKey.USER_SKILL_DB);
+  private Util.DbInfo userDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+
+  @BeforeClass
+  public static void setUp() {
+    PowerMockito.mockStatic(ServiceFactory.class);
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+    PowerMockito.mockStatic(ElasticSearchUtil.class);
+    system = ActorSystem.create("system");
+  }
+
+  @Before
+  public void beforeEachTest() {
+    PowerMockito.mockStatic(ServiceFactory.class);
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  private Request getAddSkillRequest(String userId, String endorseUserId, List<String> skillsList) {
+
+    Request actorMessage = new Request();
+    HashMap<String, Object> innerMap = new HashMap<>();
+    innerMap.put(JsonKey.REQUESTED_BY, userId);
+    actorMessage.setContext(innerMap);
+    actorMessage.put(JsonKey.ENDORSED_USER_ID, endorseUserId);
+    actorMessage.put(JsonKey.SKILL_NAME, skillsList);
+    actorMessage.setOperation(ActorOperations.ADD_SKILL.getValue());
+
+    return actorMessage;
+  }
+
+  private Request getGetSkillRequest(String userId, String endorseUserId) {
+    Request actorMessage = new Request();
+    HashMap<String, Object> innerMap = new HashMap<>();
+    innerMap.put(JsonKey.REQUESTED_BY, userId);
+    actorMessage.setContext(innerMap);
+    actorMessage.put(JsonKey.ENDORSED_USER_ID, endorseUserId);
+    actorMessage.setOperation(ActorOperations.GET_SKILL.getValue());
+    return actorMessage;
+  }
+
+  @Test
+  public void testAddSkill() {
+    Mockito.reset(cassandraOperation);
+    addSkill();
+    Mockito.reset(cassandraOperation);
+  }
+
+  private void addSkill() {
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+
+    Response response = createCassandraInsertSuccessResponse();
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(response);
+    Response response1 = createGetUserSuccessResponse();
+    when(cassandraOperation.getRecordById(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(response1);
+    subject.tell(getAddSkillRequest(USER_ID, ENDORSED_USER_ID, skillsList), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res);
+  }
+
+  @Test
+  public void testAddSkillAgain() {
+    Mockito.reset(cassandraOperation);
+    addSkill();
+    Mockito.reset(cassandraOperation);
+  }
+
+  private Response createCassandraInsertSuccessResponse() {
```

Comment:
Group test methods at one place and similarly private methods at another place. For code readability, do not interleave test methods and private methods.